### PR TITLE
Grandparent basic routes

### PR DESCRIPTION
**Pull request by RebeccaLyric**
I wanted y'all to review since the routes affect everyone's work.

### CODE REVIEW

**Comment by woodkri** [comment]
This routing looks fine to me! If it's ok with you I might refactor GrandparentReplyOpts so I can also use the /newPost endpoint.

**Comment by RebeccaLyric** [comment]
Oh, thank you for catching that Kristin! I think what we would need is a new component (something like CreatePost) into which either the GrandparentReplyOpts or FamilyReplyOpts component could be added depending on the user role.

Yes, please feel free to do any refactoring you need. I want to make sure the parts that we share are in agreement and work for both sides of the app.

**Comment by RebeccaLyric** [comment]
I had a little more time than I thought this afternoon and pushed a change to that effect. Looking at it some more, I think it would also help clarify for how my brain thinks to move related components to directories. I saw the Inbox files have their own directory and I didn't follow that pattern as I was adding things.

File: src/shared/features/CreatePost.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    // TODO: Get clicked-on post
+    let mockPost = {id: "xyz456", creatorID: "123abc", from: "Stephanie", message: "Hello, Grandpa!", photoURL: "", read: false};
+
+    return (
+        <>
+            <h2>Create Post</h2>
+            <>
+                {role === roles.poster && <div>Create a new post</div>}
+                {role === roles.receiver && <GrandparentReplyOpts/>}
```
Comment (MJGrant) [inline src/shared/features/CreatePost.tsx:35]: What is "opts" short for? Sorry, long day, out of brain cells :(

File: src/shared/features/Inbox/Inbox.tsx [hunk 2]
```diff
@@ -1,15 +1,16 @@
...
 import MailIcon from '@material-ui/icons/Mail';
 import DraftsIcon from '@material-ui/icons/Drafts';
-import { Link } from 'react-router-dom';
 
+import { Link } from 'react-router-dom';
 import { Post } from 'shared/models/post.model';
 
 import './Inbox.css';
 
+// todo: question: why here versus Inbox.css ?
```
Comment (MJGrant) [inline src/shared/features/Inbox/Inbox.tsx:13]: Material UI for React code examples always have these embedded classes that seem to be their own system independent of a .css file. I brought them over for the sake of getting things set up quickly, but I think styling belongs in the .css files (and only in the css files) and we should remove the embedded styles as we build the app. 

The system shown here hooks into theming and other stuff I haven't researched yet, though, so maybe there's merit to keeping it? We should look into it before totally dismissing it.

File: src/shared/features/CreatePost.tsx (hunk 1 above)
Comment (RebeccaLyric) [reply src/shared/features/CreatePost.tsx:35]: Oh, "options." Thanks for saying that so I know to rename and make it clearer.

File: src/shared/features/Inbox/Inbox.tsx (hunk 2 above)
Comment (RebeccaLyric) [reply src/shared/features/Inbox/Inbox.tsx:13]: I definitely agree with more research.

**Pull request merged**